include/revng/Yield/Internal/BasicBlock.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <limits>
+#include <string>
+
+#include "revng/ADT/SortedVector.h"
+#include "revng/EarlyFunctionAnalysis/BasicBlock.h"
+#include "revng/Model/VerifyHelper.h"
+#include "revng/Support/MetaAddress.h"
+#include "revng/Support/MetaAddress/YAMLTraits.h"
+#include "revng/Yield/Internal/Instruction.h"
+
+/* TUPLE-TREE-YAML
+
+name: BasicBlock
+type: struct
+fields:
+  - name: Start
+    doc: Start address of the basic block
+    type: MetaAddress
+
+  - name: End
+    doc: |
+      End address of the basic block, i.e., the address where the last
+      instruction ends
+    type: MetaAddress
+
+  - name: Successors
+    doc: List of successor edges
+    sequence:
+      type: SortedVector
+      upcastable: true
+      elementType: efa::FunctionEdgeBase
+
+  - name: Instructions
+    sequence:
+      type: SortedVector
+      elementType: yield::Instruction
+
+  - name: IsLabelAlwaysRequired
+    type: bool
+
+  - name: CommentIndicator
+    type: std::string
+  - name: LabelIndicator
+    type: std::string
```

Comment:
* [ ] What are these used for? Can you add doc?
* [ ] Drop `Indicator` suffix

Reply:
These are what indicates `Comment` and `Label` correspondingly in the official way to print assembly. For example, for most platforms `LabelIndicator` is `:`.

`CommentIndicator` varies wildly, like `@` for arm, `//` for aarch64 and so on.

2. Do you have better name ideas then? I feel like just calling them `Comment` and `Label` is going to get really confusing.

Reply:
Yeah, that was my main worry as well, but keeping track of those manually didn't felt right. This allows as to just LLVM for these, but it also means we store a whole lot of duplicates.

I thought about moving them higher up, but the highest we can possible have them in the current datastructure state is at a function level which isn't much better than the basic block level.

Do you prefer we store these in the headers somewhere then?

Reply:
In [`Architecture.h` we already have similar stuff](https://github.com/revng/revng/blob/develop/include/revng/Model/Architecture.h#L258).

Reply:
Done: https://github.com/Cvelth/revng/blob/snapshot/wip_cfg_in_svg/include/revng/Model/Architecture.h#L366.